modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotDeliveryWorker.java

```diff
@@ -421,7 +421,13 @@ public void deleteSlot(Slot slot) {
             log.debug("Releasing tracking of messages for slot " + slot.toString());
         }
         slot.deleteAllMessagesInSlot();
-        storageQueueToSlotTracker.get(slot.getStorageQueueName()).remove(slot.getId());
+
+        if (storageQueueToSlotTracker.containsKey(slot.getStorageQueueName())) {
+            storageQueueToSlotTracker.get(slot.getStorageQueueName()).remove(slot.getId());
+        } else {
```

Comment:
shall we do a storageQueueToSlotTracker.get( ...) and then check for null ...
wouldn't that be faster?